pkg/provider/docker/docker.go

```diff
@@ -109,58 +109,69 @@ type networkData struct {
 }
 
 func (p *Provider) createClient() (client.APIClient, error) {
-	host := p.Endpoint
-	var httpClient *http.Client
+	httpClient, host, err := p.getHTTPClient()
+	if err != nil {
+		return nil, err
+	}
+
+	httpHeaders := map[string]string{
+		"User-Agent": "Traefik " + version.Version,
+	}
+
+	apiVersion := DockerAPIVersion
+	if p.SwarmMode {
+		apiVersion = SwarmAPIVersion
+	}
 
-	helper, _ := connhelper.GetConnectionHelper(p.Endpoint)
+	return client.NewClientWithOpts(
+		client.WithHost(host),
+		client.WithVersion(apiVersion),
+		client.WithHTTPClient(httpClient),
+		client.WithHTTPHeaders(httpHeaders),
+	)
+}
+
+func (p *Provider) getHTTPClient() (*http.Client, string, error) {
+	helper, err := connhelper.GetConnectionHelper(p.Endpoint)
+	if err != nil {
+		return nil, "", err
```

Comment:
I think if we experience SSH URL parsing error we should continue, shouldn't we?

Reply:
@ldez looks like it can't return error if we've got correct URL. https://github.com/docker/cli/blob/8547dfcff72fd65426ef71a5bf949ac310ee3419/cli/connhelper/connhelper.go#L24

Reply:
If scheme isn't `ssh`, the only error that can be returned is error in last line, that's not defined: https://github.com/docker/cli/blob/8547dfcff72fd65426ef71a5bf949ac310ee3419/cli/connhelper/connhelper.go#L44
[url.Parse](https://godoc.org/net/url#URL.Parse) shouldn't produce errors if URL is HTTP(S) or any other. Sorry for mis-interpreting your code.